lib/pf/Authentication/Source/HTTPSource.pm

```diff
@@ -0,0 +1,266 @@
+package pf::Authentication::Source::HTTPSource;
+
+=head1 NAME
+
+pf::Authentication::Source::HTTPSource
+
+=head1 DESCRIPTION
+
+=cut
+
+use pf::constants qw($TRUE $FALSE);
+use pf::Authentication::constants;
+use pf::Authentication::Action;
+use pf::Authentication::Source;
+use URI::Escape::XS qw(uri_escape uri_unescape);
+use WWW::Curl::Easy;
+use pf::log;
+use Readonly;
+use JSON;
+
+use Moose;
+extends 'pf::Authentication::Source';
+
+has '+type' => ( default => 'HTTP' );
+has 'protocol' => ( isa => 'Str', is => 'rw', default => 'http' );
+has 'host' => ( isa => 'Str', is => 'rw', default => '127.0.0.1' );
+has 'port' => ( isa => 'Int', is => 'rw', default => '10000' );
+has 'username' => ( isa => 'Maybe[Str]', is => 'rw', default => undef );
+has 'password' => ( isa => 'Maybe[Str]', is => 'rw', default => undef );
+has 'authentication_url' => ( isa => 'Str', is => 'rw', default => '' );
+has 'authorization_url' => ( isa => 'Str', is => 'rw', default => '' );
+
+=head1 METHODS
+
+=head2 _post_curl
+
+Method used to build a basic curl object
+
+=cut
+
+sub _post_curl {
+    my ($self, $uri, $post_fields) = @_;
+    my $logger = get_logger;
+
+    $uri = $self->protocol."://".$self->host.":".$self->port.$uri;
```

Comment:
If the Authentication or Authorization URL does not have a '/' in the front then it will create a invalid url.
http://host:8080auth
Instead of 
http://host:8080/auth

So either inform the user or prepend the '/'